### PR TITLE
feat: implement delete Group

### LINKED ISSUE #218: Implement delete method of GroupRepository
## Feature Title

Implement delete method of GroupRepository after finishing https://github.com/genesis-tech-tribe/nishiki-backend/issues/215

## Acceptance Criteria

- [ ] implement delete to delete a Group and related Containers and invitationLink

### PR DESCRIPTION
**Pull request by taniguchiiqqq**
## Overviews of implementation
implement delete Group

### CODE REVIEW

File: backend/main/src/Group/Repositories/GroupRepository.ts [hunk 1]
```diff
@@ -88,10 +88,20 @@ export class GroupRepository implements IGroupRepository {
...
 	 */
 	async delete(id: GroupId): Promise<undefined> {
-		// TODO: delete all containers and users in the group.
+		// delete all containers belonging to the group.
+		const containers = await this.nishikiDbClient.listOfContainers(id.id);
+		await Promise.all(
+			containers.map((containerId) =>
+				this.nishikiDbClient.deleteContainer(containerId),
+			),
+		);
```
Comment (hitohata) [inline backend/main/src/Group/Repositories/GroupRepository.ts:101]: When a group is deleted, the user and the group relation must be deleted.
https://nishiki-tech.github.io/nishiki-documents/project-document/database#user

Comment (taniguchiiqqq) [reply backend/main/src/Group/Repositories/GroupRepository.ts:101]: Thank you for pointing it out. I totally forgot. I need to delete explicitly related data, UserGroupGSI and invitationLInk

Comment (taniguchiiqqq) [reply backend/main/src/Group/Repositories/GroupRepository.ts:101]: I found the related indexes is deleted when you delete a data in a base table. So I just added a deletion of invitation link.

> When an application writes or deletes items in a table, any global secondary indexes on that table are updated asynchronously, using an eventually consistent model.

https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GSI.html#GSI.Writes

Comment (hitohata) [reply backend/main/src/Group/Repositories/GroupRepository.ts:101]: I didn't know that information. Thank you.

**Review by hitohata — APPROVED** [review]

**Pull request merged**